lib/logging.dart

```diff
@@ -156,14 +156,14 @@ class Logger {
   /// records of different zones differently (e.g. group log records by HTTP
   /// request if each HTTP request handler runs in it's own zone).
   void log(Level logLevel, message,
-      [Object error, StackTrace stackTrace, Zone zone]) {
-    Object object;
+      [Object? error, StackTrace? stackTrace, Zone? zone]) {
+    Object? object;
     if (isLoggable(logLevel)) {
       if (message is Function) {
         message = message();
       }
 
-      String msg;
+      String? msg;
```

Comment:
This `?` shouldn't be necessary, since `msg` is definitely assigned a non-null value before it is used.  It's happening because of https://github.com/dart-lang/sdk/issues/38340, which makes the migration tool erroneously conclude that line 168 might store a null value in `msg`.